### ISSUE TITLE
Negative return code from AdsSyncWriteReqEx

### ISSUE THREAD

**Issue by satether** [Beckhoff/ADS#238]
My project at work is using v20 of this library to enable an AlmaLinux 9.4 system to communicate with a Beckhoff controller running TwinCAT 3.1.4022. Once every few days or so an attempt to write to a PLC variable with AdsSyncWriteReqEx() fails with a return code of -1. That's not a valid ADS code, so we thought it might be a negated errno code. Is that right or is it something else?

**Comment by satether** [comment]
This problem appears to have been caused by the unexpected closure of the network connection between the Linux system and the Beckhoff controller. As a result the next attempt to send an ADS command fails as I described, probably due to the fd being invalid. The closure looks like it was caused by our experimenting with different ways to transmit PTP packets to the EL6688 terminal om out EtherCAT bus. I guess one can't tamper with time with impunity!

**Comment by satether** [comment]
We eventually traced our problem to this: Our system is controlled with production software assigned a local ADS address. We were also using adstool to read out diagnostic CoE information and to change some settings in CoE. adstool had to run on the same host as the production software and ended up using the same local ADS address. Once adstool exited, the production software's connection to our device was also severed, every time. That's when we started getting return codes of -1 from ADS calls in the production software.
